### DISCUSSION TITLE
How to stream live multiple RTSP source to My HLS Player

### DISCUSSION THREAD

**Discussion by deorwine** [bluenviron/mediamtx#3349] [Questions]
### Question

How i can stream Live rtsp source to hls player using mediamtx and also want to save it on s3 . ? i have not found direct stream to web can you please suggest i am going to use mediamtx first time?

**Comment by aler9** [comment]
Hello, if you want to stream a RTSP source to a HLS player it's enough to ingest the RTSP source into the server by following instructions in the README:
https://github.com/bluenviron/mediamtx?tab=readme-ov-file#rtsp-cameras-and-servers

Conversion to HLS is automatic, therefore the HLS stream will be available here:
https://localhost:8888/rtsp_source_name

Replace `rtsp_source_name` with the name of the configuration you created in the previous step.

Regarding the ability to store HLS streams into S3, we don't support it since external HTTP servers are not capable of of serving an HLS low-latency stream, since they miss the ability to provide [blocking playlists](https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-6.2.5.2) and [preload hints](https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-6.2.6). If you want to scale, you can use a CDN in front of the server, as mentioned in the README.

Nonetheless, we support uploading streams to S3 for recording purposes, in MPEG-TS or fMP4 format. Instructions are in the README:
https://github.com/bluenviron/mediamtx?tab=readme-ov-file#record-streams-to-disk

[a comment by an automated account is not shown]
